src/Workspaces/Core/Portable/Diagnostics/DiagnosticDataLocation.cs

```diff
@@ -31,104 +31,72 @@ internal sealed class DiagnosticDataLocation
         [DataMember(Order = 2)]
         public readonly TextSpan? SourceSpan;
 
-        [DataMember(Order = 3)]
-        public readonly int OriginalStartLine;
-
-        [DataMember(Order = 4)]
-        public readonly int OriginalStartColumn;
-
-        [DataMember(Order = 5)]
-        public readonly int OriginalEndLine;
-
-        [DataMember(Order = 6)]
-        public readonly int OriginalEndColumn;
-
         /// <summary>
-        /// Null if path is not mapped and <see cref="OriginalFilePath"/> contains the actual path.
-        /// Note that the value might be a relative path. In that case <see cref="OriginalFilePath"/> should be used
-        /// as a base path for path resolution.
+        /// Null if path is not mapped and <see cref="OriginalFileSpan"/> contains the actual path. Note that the value
+        /// might be a relative path. In that case <see cref="OriginalFileSpan"/> should be used as a base path for path
```

Comment:
:( - is it possible to force the conversion on the creator - aka if they have a relative mapped file path they are responsible for making it not relative?

Reply:
Yes.  I think so.  I will make a followup PR to try that approach out.  Thanks!